mbed_flasher/flashers/FlasherST.py

```diff
@@ -94,6 +97,17 @@ def flash(self, source, target, method, no_reset, target_filename=None):
         :param target_filename: target filename (not in use)
         :return: 0 when flashing success
         """
+        tmp_file = None
+        try:
+            if target_filename:

```

Comment:
This code, of course, adds more defect surface and lowers the test coverage...
However, this whole `FlasherST.py` goes to trash anyway in the future